src/lib.rs

```diff
@@ -134,6 +134,8 @@
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
 // Coding conventions
 #![deny(missing_docs)]
+#![allow(clippy::suspicious_arithmetic_impl)] // this lint is literally always wrong
```

Comment:
No, it's wrong even when doing other kinds of arithmetic. This lint has fired in multiple projects I have worked on and it is literally always wrong.

One time I actually did have a bug where I was using the wrong operator in an operator override impl, and the lint **didn't** fire because I hit some sort of obscure false negative. It's pathetic.

Anyway conceptually the lint is confused. If I was implementing `+` only in terms of other `+` operators I'd hardly need a new type, would I? Or is the argument that Rust has such an anemic set of operator that `+` is used for everything and I shouldn't ever be using non-`+`? That would track but it's not what the lint says.

Reply:
Warning, I being a massive knob right now.

From the name I assumed it was to catch impls on simple wrapper types where one did a cut'n'pasta and used `+` in `ops::Sub` e.g. the lint fires here as expected 

```rust
use std::ops;

fn main() {
    let one = Foo(1);
    let two = Foo(2);
    let three = Foo(3);

    assert!(three - two == one)
}

#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
struct Foo(u64);

impl ops::Add for Foo {
    type Output = Self;
    fn add(self, rhs: Self) -> Self::Output { Foo(self.0 + rhs.0) }
}

impl ops::Sub for Foo {
    type Output = Self;
    fn sub(self, rhs: Self) -> Self::Output { Foo(self.0 + rhs.0) }
}
```

```bash
% cargo clippy
    Checking play v0.1.0 (/home/tobin/build/play)
warning: suspicious use of `+` in `Sub` impl
  --> src/main.rs:21:58
   |
21 |     fn sub(self, rhs: Self) -> Self::Output { Foo(self.0 + rhs.0) }
   |                                                          ^
```

Reply:
> // this lint is literally always wrong

So not _always_ :)

Reply:
Fine. I will change the text to say "this lint is conceptually confused to the point where *at best* it could be saying "double-check this operator impl" on a subset of impls that are more likely to be wrong than the set of all operator impls. It is plausible that, when taken over the entire corpus of Rust code, this could be true. But when taken over the corpus of code written by rust-bitcoin maintainers, it fails even this condition. Instead it fires on code which is correct and never fires on code that is incorrect."

Maybe I should make a new module just for this comment?

Reply:
I am being sarcastic above, of course.

But I would be willing to remove the word "literally" because in the case below (where I was using `+` for `Polynomial::sub` instead of passing through to the underlying field subtraction) it would have been accidentally correct. But the code in question was not copy-pasted, was not a "dumb wrapper type", and it was surrounded by other impls that were correct but also would have triggered the lint.

Reply:
Well, let me sleep on it. I hate this lint so much. It's just rubbing sand in the face of everybody who wanted a sane set of operators for this language. It doesn't complain when people use `+` for concatenation but it does when people use it for pretty-much anything that somebody might describe as "addition".

Reply:
\me chuckles

I'm just that guy trying to save us from future hate from the whole entire Rust community :)

Reply:
The existing text is already far more polite than is warranted.

I can remove the word "literally" if you believe it is technically incorrect (though I note your only evidence is hypothetical code that you assume the lint is for). But that is as far as I am willing to go in the direction of weakening my claim.

Reply:
lol, love it. Lets just leave it as is :)